### PR TITLE
feat: block id kwarg support [APE-1240]

### PR DESCRIPTION
**Pull request by antazoey**
### What I did

fixes: #1573 
Fixes: APE-1240

### How I did it

both kwargs

### How to verify it

use kwarg

### Checklist

- [ ] All changes are completed
- [ ] New test cases have been added
- [ ] Documentation has been updated

### CODE REVIEW

File: src/ape/api/providers.py [hunk 1]
```diff
@@ -1100,7 +1100,7 @@ def _prepare_call(self, txn: TransactionAPI, **kwargs) -> List:
         txn_dict.pop("maxFeePerGas", None)
         txn_dict.pop("maxPriorityFeePerGas", None)
 
-        block_identifier = kwargs.pop("block_identifier", kwargs.pop("block_id", "latest"))
```
Comment (fubuloubu) [inline src/ape/api/providers.py:1103]: What's wrong with `"latest"`?

**Review by fubuloubu — APPROVED** [review]

File: src/ape/api/providers.py (hunk 1 above)
Comment (antazoey) [reply src/ape/api/providers.py:1103]: nothing, besudes it's the default already when passing `None` to web3, so less management on our end by using `None`.

File: src/ape_test/provider.py [hunk 2]
```diff
@@ -77,7 +77,7 @@ def estimate_gas_cost(self, txn: TransactionAPI, **kwargs) -> int:
             block = self.web3.eth.get_block("latest")
             return block["gasLimit"]
 
-        block_id = kwargs.pop("block_identifier", None)
+        block_id = kwargs.pop("block_identifier", kwargs.pop("block_id", None))
```
Comment (fubuloubu) [inline src/ape_test/provider.py:80]: Here too?

**Review by fubuloubu — APPROVED** [review]

File: src/ape_test/provider.py (hunk 2 above)
Comment (antazoey) [reply src/ape_test/provider.py:80]: if it was `None` before this PR, it doesnt need to change

**Pull request merged**